### PR TITLE
Fix the `throwHttpErrors` option for POST requests

### PR DESCRIPTION
**Pull request by daixianceng**
Here is a bug:

When I set `throwHttpErrors` to `false` and then `POST` a request, if the response is not OK, it will still throw `HTTPError`. So I fixed it.

### CODE REVIEW

**Comment by sindresorhus** [comment]
Can you add a test?

**Comment by daixianceng** [comment]
All tests passed

File: index.js [hunk 1]
```diff
@@ -183,12 +184,14 @@ class Ky {
 					}
 				}
 
-				if (!response.ok) {
+				if (!response.ok && (isRetry || (!isRetry && throwHttpErrors))) {
```
Comment (jenbuzz) [inline index.js]: It works as is but using "this.throwHttpErrors" would be more in-sync with the rest of the code.

```suggestion
				if (!response.ok && (isRetry || (!isRetry && this.throwHttpErrors))) {
```

File: index.js [hunk 2]
```diff
@@ -167,7 +167,8 @@ class Ky {
 		this._response = this._fetch();
 
 		for (const type of responseTypes) {
-			this._response[type] = this._retry(async () => {
+			const isRetry = retryMethods.has(this._options.method.toLowerCase());
```
Comment (jenbuzz) [inline index.js]: isRetry might be a little confusing since it's more like a "allowRetry" check

File: index.js (hunk 1 above)
Comment (sholladay) [reply index.js]: Saying `isRetry || (!isRetry && throwHttpErrors)` is the same as just saying `isRetry || throwHttpErrors`

File: index.js (hunk 2 above)
Comment (sholladay) [reply index.js]: Maybe `isRetriableMethod`?

**Comment by sindresorhus** [comment]
Thanks :)

**Pull request merged**